samples/subsys/lorawan/class_a/overlay-multicast.conf

```diff
@@ -0,0 +1,9 @@
+# NVS required to store LoRaWAN DevNonce and multicast sessions
+CONFIG_NVS=y
+CONFIG_SETTINGS=y
+CONFIG_LORAWAN_NVM_SETTINGS=y
```

Comment:
Is this required? The multicast code doesn't do anything extra with settings from what I can see.

Reply:
Yes, it is required. The multicast keys are stored in NVM. Backends expect multicast sessions to be persistent once set up.

This is also the reason why multicast depends on `LORAWAN_NVM_SETTINGS`.

Reply:
Expecting it to be persistent is different from requiring `LORAWAN_NVM_SETTINGS`. For example I use a NVS based backend for LoRaWAN state. I imagine that would work fine, but is not accurately expressed by `depends on LORAWAN_NVM_SETTINGS`. I think `depends on !LORAWAN_NVM_NONE` is more accurate?

Reply:
That means you have added a custom backend that would also store *all* settings from the LoRaMAC-node stack and not only the DevNonce? I can switch the dependency to `!LORAWAN_NVM_NONE`, but for the in-tree sample we still need the settings, right?

Reply:
Updated Kconfig dependencies to:

```
	depends on LORAWAN_NVM
	depends on !LORAWAN_NVM_NONE
```